packages/types/src/search.ts

```diff
@@ -31,3 +35,26 @@ export type SearchParameter = {
     per_page: number;
     include_deleted_channels: boolean;
 }
+
+export type SearchParams = {
```

Comment:
Are the types of these fields correct? Based on the server's definitions ([link](https://github.com/mattermost/mattermost-server/blob/master/model/search_params.go#L16)), all of the fields should be optional and it looks like channels and users are just their IDs instead of whole objects

Reply:
Yes, you are right! all the fields are optional, however this type corresponds to the output of of the selector `selectRecentSearches` where `excluded_channels` and `in_channels` are mapped from channel ids to objects of type `Channel` and if the field is empty it's value is an empty array, same thing for `from_users` and `excluded_users` where they are mapped to `UserProfile` objects.

The mapping to objects is needed to be able to format the recent search to a readable format for the user, example:
```
included_users: ["id1"] (recieved) -> [{id: "id1", username: "test-user", nickname: "Testing user",...}] (mapped to)
and
included_users: null -> []
```

Reply:
Oh, I see. I didn't notice that those fields were mapped like that. Code-wise, that makes sense to me to do, but I would like a separate type added then like you mentioned so that it's clear that it's a different type and to avoid any future potential naming conflicts. If you just want to rename this one to `RecentSearchParams`, I think that'd work well enough.

I also just noticed that the server has a `model.SearchParams` type (corresponding closer to this) and a `model.SearchParameter` type (corresponding to the above `SearchParameter` type) for whatever reason, but since your new type matches `model.SearchParams` closer, I think it makes sense to keep the naming that way. The naming is rather confusing, but let's just be consistent with the server here. From what I can tell, `SearchParameter` is what is sent to the server which is then parsed into a `SearchParams` object.

Reply:
That will do. I agree, having separate types is clearer.